src/lava/proc/conv/process.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from lava.proc.conv import utils
+
+
+class Conv(AbstractProcess):
+    """Convolution connections between neurons.
+
+    Parameters
+    ----------
+    input_shape : tuple of three ints
+        shape of input to the process in (X, Y, Z) or (W, H, C) format.
+    weight : tensor/array
+        convolution kernel weight. The dimension should be in
+        (C_out, W, H, C_in) format.
+    padding : int or tuple of two ints
+        convolution padding size. Default is 0.
+    stride : int or tuple of two ints
+        convolution stride. Default is 1.
+    dilation : int or tuple of two ints
+        convolution dilation. Default is 1.
+    groups : int
+        number of groups in convolution. Default is 1.
+
+    Note
+    ----
+    padding, stride and dilation are expected in (X, Y) or (W, H) if tuple.
+    """
+    def __init__(self, **kwargs):
+        # the process In/OutPort shapes are considered to be XYZ(WHC) format.
```

Comment:
Sentences usually start with a capital letter.

Reply:
That is one of my bad habits... I usually do that in comments :(

Reply:
That's one of my bad habit. I tend to do that in comments.